server/src/main/java/org/elasticsearch/action/admin/cluster/settings/ClusterUpdateSettingsResponse.java

```diff
@@ -103,7 +104,22 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
-    public static ClusterUpdateSettingsResponse fromXContent(XContentParser parser) throws IOException {
+    public static ClusterUpdateSettingsResponse fromXContent(XContentParser parser) {
         return PARSER.apply(parser, null);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (super.equals(o)) {
```

Comment:
I would like to double check this, I'm not sure if using super might violate the equals() contract.

Reply:
I think its all good, because AcknowledgedResponse is abstract. If it would be possible to create instances of it, I think there could have been cases where the symmetry property of equals() would be broken, but this cannot happen with non-instantiable superclasses so I think this is all good.